docs/busy-throttling.md

```diff
@@ -0,0 +1,26 @@
+# Throttling on Busy Responses
+
+In TChannel, a busy response is sent to indicate that a service is overloaded. Upon
+receiving busy responses, the client should reduce its requests to the service so
+that the service can recover to a healthy state.
+
+In order to do throttling, it is necessary to measure how fast the client is sending
+the requests. We use RPS (Request Per Second), i.e., the number of request sent every
+second. Given the client `clientA` that sends requests to the sevice `serviceB` with a
+RPS of `RpsToB`, there are three scenairos that we care about:
+
+1. All requests responeded without any busy responses. In this case, no request
+   should be throttled. `clientA` should continue to send all requests to `serviceB`. 
+2. If a busy response is received, `clientA` should throttle its sent requests to `serviceB`.
```

Comment:
this is similar in many ways to TCP congestion control, busy response is like packet drop due to congestion, throttling rate is like packets window size:
https://en.wikipedia.org/wiki/TCP_congestion-avoidance_algorithm

CUBIC is used by default in the latest Linux kernel, in which the window is a cubic function of time since the last congestion event:
https://en.wikipedia.org/wiki/CUBIC_TCP

my feeling is cutting by half upon congestion might be too aggressive, and increasing rate by 5% might be not enough for the traffic to ramp up quickly again.